blosc2/CMakeLists.txt

```diff
@@ -18,13 +25,9 @@ else()
     set(BLOSC_INSTALL ON)
     add_subdirectory(c-blosc2)
     include_directories("${CMAKE_CURRENT_SOURCE_DIR}/c-blosc2/include")
-    target_link_libraries(blosc2_ext blosc2_static)
+    target_link_libraries(blosc2_ext PRIVATE blosc2_static)
```

Comment:
The `c-blosc2` would need a through review to bring it in line with modern CMake:
- Change `include_directories` etc. to their target scoped counterparts (`target_include_directories`)
- Create aliases and namespace the options to make it consumable by `FetchContent`
- Do not create separate shared/static targets, there are simpler alternatives

Reply:
I'm a not a cmake guru, so I'd appreciate a PR from more knowledgeable people for modernizing it in upstream `c-blosc2`.